test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -15607,6 +15607,38 @@ BOOST_AUTO_TEST_CASE(contract_name)
 	ABI_CHECK(callContractFunction("constantNameAccessor()"), argsLong);
 }
 
+BOOST_AUTO_TEST_CASE(event_wrong_abi_name)
+{
+	char const* sourceCode = R"(
+		contract Another {
+			uint xx;
+			function abc() public {}
+		}
+		library ClientReceipt {
+			event Deposit(Another indexed _from, bytes32 indexed _id, uint _value);
+			function deposit(bytes32 _id) public {
+				Another a;
+				emit Deposit(a, _id, msg.value);
+			}
+		}
+		contract Test {
+			function f() public {
+				ClientReceipt.deposit("123");
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "Another");
+	compileAndRun(sourceCode, 0, "ClientReceipt", bytes(), map<string, Address>{{"Another", m_contractAddress}});
```

Comment:
`Another` is not a library, so you do not need to provide linking information here.